src/olympia/addons/models.py

```diff
@@ -1260,15 +1260,18 @@ def is_unreviewed(self):
     def is_public(self):
         return self.status == amo.STATUS_PUBLIC and not self.disabled_by_user
 
-    def has_complete_metadata(self):
-        return all(self.get_required_metadata())
-
-    def get_required_metadata(self):
-        if not self.has_listed_versions():
+    def has_complete_metadata(self, for_channel=None):
+        return all(self.get_required_metadata(for_channel=for_channel))
+
+    def get_required_metadata(self, for_channel=None):
+        if for_channel is None:
+            for_channel = (
+                amo.RELEASE_CHANNEL_LISTED if self.has_listed_versions()
+                else amo.RELEASE_CHANNEL_UNLISTED)
+        if for_channel == amo.RELEASE_CHANNEL_UNLISTED:
             # Add-ons with only unlisted versions have no required metadata.
             return []
-        latest_version = self.find_latest_version(
-            channel=amo.RELEASE_CHANNEL_LISTED)
+        latest_version = self.versions.latest()  # Version.from_upload does it.
```

Comment:
The filtering in `get_required_metadata` was wrong (it would produce a falsey result in step 4) compared to how `Version.from_upload` works now.  As for whether `Version.from_upload` _should_ copy from to a new unlisted version, I'm of two minds - yes, it is wrong to copy a license when it's not needed; but it does allow the scenario to work and not force the developer to supply it again (and allow them to use the API for the upload).